etw/event.go

```diff
@@ -0,0 +1,87 @@
+package etw
+
+// Channel represents the ETW logging channel that is used. It can be used by
+// event consumers to give an event special treatment.
+type Channel uint8
+
+const (
+	// ChannelTracelogging is the default channel for tracelogging events. It is
+	// not required to be used for tracelogging, but will prevent decoding
+	// issues for these events on older operating systems.
+	ChannelTracelogging Channel = 11
+)
+
+// Level represents the ETW logging level. There are several predefined levels
+// that are commonly used, but technically anything from 0-255 is allowed.
+// Lower levels indicate more important events, and 0 indicates an event that
+// will always be collected.
+type Level uint8
+
+// Predefined ETW log levels.
+const (
+	LevelAlways Level = iota
+	LevelCritical
+	LevelError
+	LevelWarning
+	LevelInfo
+	LevelVerbose
+)
+
+// Event represents a single ETW event. It can have field metadata and data
+// added to it, and then be logged via a provider to actually send it to ETW.
+type Event struct {
+	Descriptor *EventDescriptor
+	Metadata   *EventMetadata
+	Data       *EventData
+}
+
+// NewEvent returns a new instance of an event object.
+func NewEvent(name string, descriptor *EventDescriptor) *Event {
+	return &Event{
+		Descriptor: descriptor,
+		Metadata:   NewEventMetadata(name),
+		Data:       &EventData{},
+	}
+}
+
+// EventDescriptor represents various metadata for an ETW event.
+type EventDescriptor struct {

```

Comment:
Since you broke the others up do you want `EventDescriptor` in its own file as well?

Reply:
I mostly moved EventData and EventMetadata to their own files because I anticipate their set of functions growing over time as we support for data types. EventDescriptor should stay small, so I'm okay with it staying in this file.